From b282151dd359ad847624a79768ebef7af71e6def Mon Sep 17 00:00:00 2001
From: Kexy Biscuit <kexybiscuit@aosc.io>
Date: Fri, 26 Jul 2024 17:09:46 +0800
Subject: [PATCH 30/31] Move chrome/browser/ui:webui_name_variants to
 public_deps

The chrome/browser/ui build target contains webui_contents_wrapper.h,
which included the generated webui_name_variants.h header. For this to
work correctly, the build system must process the webui_name_variants
target first and generate webui_name_variants.h. To do this reliably,
without having to depend on webui_name_variants in all targets that
transitively include webui_contents_wrapper.h, make it a public_deps.

Bug: 40253918, 346711540, 346699817, 345645751
Change-Id: I6e0b573427cda1b2b408702f139b2c9e83f05045

https://chromium-review.googlesource.com/c/chromium/src/+/5646245/1

Co-authored-by: Lei Zhang <thestig@chromium.org>
---
 chrome/browser/ui/BUILD.gn | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/chrome/browser/ui/BUILD.gn b/chrome/browser/ui/BUILD.gn
index 49e6bdcbd5..3d1ccee5c6 100644
--- a/chrome/browser/ui/BUILD.gn
+++ b/chrome/browser/ui/BUILD.gn
@@ -6367,7 +6367,6 @@ static_library("ui") {
     sources += get_target_outputs(":webui_name_variants")
 
     deps += [
-      ":webui_name_variants",
       "side_search:side_search_tab_data_proto",
       "//base",
       "//chrome/browser:main_extra_parts",
@@ -6426,7 +6425,10 @@ static_library("ui") {
       "//ui/lottie",
       "//ui/views:buildflags",
     ]
-    public_deps += [ "//ui/base/dragdrop/mojom:mojom_headers" ]
+    public_deps += [
+      ":webui_name_variants",
+      "//ui/base/dragdrop/mojom:mojom_headers",
+    ]
 
     allow_circular_includes_from += [ "//chrome/browser/ui/views" ]
 
-- 
2.45.2

